xcube/core/gridmapping/helpers.py

```diff
@@ -50,7 +50,7 @@ def _to_int_or_float(x: Number) -> Number:
         return x
     xi = int(x)
     xf = float(x)
-    return xi if math.isclose(xi, xf, abs_tol=1e-10) else xf
+    return xi if math.isclose(xi, xf, abs_tol=1e-5) else xf
```

Comment:
```suggestion
    return xi if math.isclose(xi, xf, rel_tol=1e-5) else xf
```

I'm pretty sure, `rel_tol` is more correct: https://docs.python.org/3/library/math.html#math.isclose